src/io/flutter/dart/FlutterDartAnalysisServer.java

```diff
@@ -75,22 +91,79 @@ private void removeSubscription(@NotNull final String service, @NotNull final St
   }
 
   private void sendSubscriptions() {
-      final String id = analysisService.generateUniqueId();
-      analysisService.sendRequest(id, FlutterRequestUtilities.generateAnalysisSetSubscriptions(id, subscriptions));
+    final String id = analysisService.generateUniqueId();
+    analysisService.sendRequest(id, FlutterRequestUtilities.generateAnalysisSetSubscriptions(id, subscriptions));
   }
 
   @NotNull
   public List<SourceChange> edit_getAssists(@NotNull VirtualFile file, int offset, int length) {
-      return analysisService.edit_getAssists(file, offset, length);
+    return analysisService.edit_getAssists(file, offset, length);
+  }
+
+  @Nullable
+  public SourceChange flutter_getChangeAddForDesignTimeConstructor(@NotNull VirtualFile file, int _offset) {
+    final String filePath = FileUtil.toSystemDependentName(file.getPath());
+    final int offset = getOriginalOffset(file, _offset);
+
+    final CountDownLatch latch = new CountDownLatch(1);
+    final AtomicReference<SourceChange> result = new AtomicReference<>();
+    final String id = analysisService.generateUniqueId();
+    responseConsumers.put(id, (resultObject) -> {
+      try {
+        final JsonObject changeObject = resultObject.getAsJsonObject("change");
+        final SourceChange change = SourceChange.fromJson(changeObject);
+        result.set(change);
+      }
+      catch (Throwable ignored) {
+      }
+      latch.countDown();
+    });
+
+    final JsonObject request = FlutterRequestUtilities.generateFlutterGetChangeAddForDesignTimeConstructor(id, filePath, offset);
+    analysisService.sendRequest(id, request);
+
+    Uninterruptibles.awaitUninterruptibly(latch, 100, TimeUnit.MILLISECONDS);
```

Comment:
This is long enough (for a busy server)?

Reply:
Hm... A busy server might be quite slow at time.
We use this delay for requesting quick assist.